plugin/hooks.go

```diff
@@ -50,31 +50,43 @@ type Hooks interface {
 	// OnActivate is invoked when the plugin is activated. If an error is returned, the plugin
 	// will be terminated. The plugin will not receive hooks until after OnActivate returns
 	// without error. OnConfigurationChange will be called once before OnActivate.
+	//
+	// Minimum server version: 4.2
```

Comment:
Ah, this is fascinating. I'd forgotten that this exact file existed prior to v5.2 even though we completely rewrote the plugin API in that release. My apologies for the confusion here, but the minimum server version for /now/ should be v5.2. See https://developers.mattermost.com/extend/plugins/migration/#migrating-plugins-from-mattermost-51-and-earlier for the details. Can you bump accordingly?

@crspeller / @hanzei, any concerns with that?

Reply:
Agreed. Minimums should all be >v5.2

Reply:
Perfect, so I will change all the mins under version 5.2 for 5.2 @lieut-data @crspeller

Reply:
Done, basically, they are all 5.2 except `UserHasBeenCreated` that is 5.10.